packages/kit/src/vite/build/utils.js

```diff
@@ -150,3 +150,14 @@ export function remove_svelte_kit(config) {
 		.flat(Infinity)
 		.filter((plugin) => plugin.name !== 'vite-plugin-svelte-kit');
 }
+
+// If we'd written this in TypeScript, it could be easy...
+/**
+ * @param {string} maybeHttpMethod
+ * @returns {maybeHttpMethod is import('types').HttpMethod}
+ */
+export function isHttpMethod(maybeHttpMethod) {
+	const method_names = ['GET', 'HEAD', 'PUT', 'POST', 'DELETE', 'PATCH'];
+
+	return method_names.includes(maybeHttpMethod);
+}
```

Comment:
the return value is just a boolean, we don't need anything fancy. most of the time we let TypeScript infer return values, unless it's a longer function or the return value is particularly complex/inscrutable

Reply:
I'm not home right now, but when I was making this commit, I actually just started with `.filter(export => method_names.includes(export))`. However, that ends up returning the same type as the original array (`Array<string>`). You've then got two options: Use a JSDoc cast to just force-cast the result to the correct type, or use a type predicate as the filter argument. The proposed solution (using a boolean return type) actually causes an error for `pnpm check` on my end.

Reply:
ooooohhhhhhh i see! i'm a simple man and that flew right over my head

Reply:
Yeah, confirmed -- changing return type to inferred means `lookup[file].filter(is_http_method)` is inferred as an `Array<string>`, which is not assignable to `methods[file]`, whereas it's an `Array<HttpMethod>` if you specify the return type of `is_http_method` is `str is HttpMethod` rather than just `bool`.